docs/contributing-to-airbyte/tutorials/building-a-java-destination.md

```diff
@@ -0,0 +1,175 @@
+
+**TDD using acceptance tests & integration tests**
+
+Airbyte provides a standard test suite (dubbed "Acceptance Tests") that is run against every destination. The objective of these tests is to provide some "free" baseline tests that ensure that the basic functionality of the destination works. One approach to developing your connector is to simply run the tests between each change and use the feedback from them to guide your development.
+
+If you want to try out this approach, check out Step 6 which describes what you need to do to set up the acceptance Tests for your destination.
+
+The nice thing about this approach is that you are running your destination exactly as Airbyte will run it in the CI. The downside is that the tests do not run very quickly. As such, we recommend this iteration approach only once you've implemented most of your connector and are in the finishing stages of implementation. 
+
+**Directly running the destination using Docker**
+
+If you want to run your destination exactly as it will be run by Airbyte \(i.e. within a docker container\), you can use the following commands from the connector module directory \(`airbyte-integrations/connectors/destination-<name>`\):
+
+```text
+# First build the container
+./gradlew :airbyte-integrations:connectors:destination-<name>:build
+
+# Then use the following commands to run it
+# Runs the "spec" command, used to find out what configurations are needed to run a connector 
+docker run --rm airbyte/destination-<name>:dev spec
+
+# Runs the "check" command, used to validate if the input configurations are valid
+docker run --rm -v $(pwd)/secrets:/secrets airbyte/destination-<name>:dev check --config /secrets/config.json
+
+# Runs the "write" command which reads records from stdin and writes them to the underlying destination
+docker run --rm -v $(pwd)/secrets:/secrets -v $(pwd)/sample_files:/sample_files airbyte/destination-<name>:dev write --config /secrets/config.json --catalog /sample_files/configured_catalog.json
+```
+
+Note: Each time you make a change to your implementation you need to re-build the connector image via `./gradlew :airbyte-integrations:connectors:destination-<name>:build`.
+
+The nice thing about this approach is that you are running your destination exactly as it will be run by Airbyte. The tradeoff is that iteration is slightly slower, because you need to re-build the connector between each change.
+
+### Step 3: Implement `spec`
+
+Each destination contains a specification that describes what inputs it needs in order to run. Defining the specification is a good place to start when developing your destination. The easiest way is to use [JsonSchema](https://json-schema.org/) to define what the inputs are \(e.g. username and password\). Here's [an example](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/destination-postgres/src/main/resources/spec.json) of what the `spec.json` looks like for the postgres destination.
```

Comment:
It feels useful to keep the reference to JsonSchema in case people need to check the syntax. Maybe something like this:

```suggestion
Each destination contains a specification written in JsonSchema that describes its inputs. Defining the specification is a good place to start when developing your destination. Check out the documentation [here](https://json-schema.org/) to learn the syntax. Here's [an example](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/destination-postgres/src/main/resources/spec.json) of what the `spec.json` looks like for the postgres destination.
```